server/socket.js

```diff
@@ -54,21 +55,22 @@ module.exports = io => {
       console.log(chalk.magenta(`${socket.id} has disconnected`));
       leaveChatRoom();
       console.log(`[${socket.id}] disconnected`);
-      delete sockets[socket.id];
+      store.dispatch(removeSocket(socket));
     });
 
     // joinChatRoom joins a socket.io room and tells all clients in that room to establish a WebRTC
     //   connetions with the person entering the room.
     socket.on('joinChatRoom', function (room) {
       console.log(`[${socket.id}] join ${room}`);
-      if (!(room in rooms)) {
-        rooms[room] = {};
-      }
-      for (const id in rooms[room]) {
-        rooms[room][id].emit('addPeer', { 'peer_id': socket.id, 'should_create_offer': false });
-        socket.emit('addPeer', { 'peer_id': id, 'should_create_offer': true });
+      if (!(store.getState().rooms.has(room))) {
+        console.log(`Adding ${room} to state`);
+        store.dispatch(addRoom(room));
       }
-      rooms[room][socket.id] = socket;
+      store.getState().rooms.get(room).valueSeq().forEach(peer => {
```

Comment:
To make this (& line 86) slightly more consumable (???), I'd maybe save part of it as a variable, but I also understand that that variable wouldn't be used anywhere else sooo...

...idk I'm conflicted. Not really asking you to change anything. Just expressing the feelz.